### PR TITLE
Add an intro to fuzzing document.

### PR DESCRIPTION
**Pull request by mbarbella-chromium**
PTAL. If you can think of any larger sections that should be added, let me know and I'll be happy to add them.

### CODE REVIEW

File: docs/intro-to-fuzzing.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+For example, if a browser defines a function that takes an input string of CSS
+and parses it, that may be a good candidate for fuzzing. It’s a clear entry
+point that could be used for a fuzzer, and it’s likely that arbitrary
+attacker-supplied input would be handled by it as well.
+
+If code is exposed to the network, it is usually safe to assume it expects to
+handle untrusted inputs. Though the same advice from above applies, some fuzzing
+tools allow this code to be
+[tested directly](https://github.com/google/honggfuzz/blob/master/socketfuzzer/README.md).
```
Comment (oliverchang) [inline docs/intro-to-fuzzing.md:63]: how about a sentence on correctness fuzzing and point to some examples?

File: docs/intro-to-fuzzing.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+[libfuzzer]: https://llvm.org/docs/LibFuzzer.html
+
+### [Honggfuzz]
+
+[Honggfuzz] is another fuzzing tool with a rich set of
+[features](https://github.com/google/honggfuzz#features) available to it. It
+does not necessarily expect targets to be defined and formed in the same style
+as libFuzzer-based targets, though it can be used in the same way.
+
+Application developers that require more customization in how input is fed to
```
Comment (oliverchang) [inline docs/intro-to-fuzzing.md:93]: can you expand on this a bit? it's a bit unclear what exactly honggfuzz can offer here.

File: docs/intro-to-fuzzing.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+```cpp
+// fuzz_target.cc
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+  DoSomethingInterestingWithMyAPI(Data, Size);
+  return 0;  // Non-zero return values are reserved for future use.
+}
+```
+
+For new projects setting up fuzzing for the first time, we tend to recommend
```
Comment (oliverchang) [inline docs/intro-to-fuzzing.md:80]: It seems a bit confusing which one someone new to fuzzing should pick. 

Probably make this point stronger about how libFuzzer is the most friendly to use for developers, and should be picked over others. AFL/honggfuzz might be more useful at first to security folks. 

Maybe also talk about how each engine has its own benefits and link to fuzzbench under further reading or somewhere similar?

**Review by oliverchang — APPROVED** [review]

**Review by inferno-chromium — APPROVED** [review]

File: docs/intro-to-fuzzing.md (hunk 2 above)
Comment (mbarbella-chromium) [reply docs/intro-to-fuzzing.md]: Done!

**Pull request merged**